src/rest/connector/libs/xpresso/implementation.py

```diff
@@ -0,0 +1,310 @@
+import json
+import logging
+import requests
+import os
+from requests.exceptions import RequestException
+from pyats.connections import BaseConnection
+from rest.connector.implementation import Implementation as Imp
+from rest.connector.utils import get_token
+
+# create a logger for this module
+log = logging.getLogger(__name__)
+
+
+class Implementation(Imp):
+    '''Rest Implementation for xpresso
+
+    Implementation of Rest connection to devices based on pyATS BaseConnection
+    for xpresso
+
+    YAML Example
+    ------------
+
+        devices:
+            xpresso:
+                os: xpresso
+                connections:
+                    rest:
+                        class: rest.connector.Rest
+                        host : "xpresso.cisco.com"
+                        credentials:
+                            rest:
+                                token: <xpressoaccesstoken>
+
+    Code Example
+    ------------
+
+        >>> from pyats.topology import loader
+        >>> testbed = loader.load('testbed.yaml')
+        >>> device = testbed.devices['xpresso']
+        >>> device.connect(alias='rest', via='rest')
+        >>> device.rest.connected
+        True
+    '''
+    @BaseConnection.locked
+    def connect(self, timeout=30):
+        '''connect to the device via REST
+
+        Arguments
+        ---------
+
+            timeout (int): Timeout value
+
+        Raises
+        ------
+
+        Exception
+        ---------
+
+            If the connection did not go well
+
+        Note
+        ----
+
+        There is no return from this method. If something goes wrong, an
+        exception will be raised.
+
+
+        YAML Example
+        ------------
+        devices:
+            xpresso:
+                os: xpresso
+                connections:
+                    rest:
+                        class: rest.connector.Rest
+                        host : "xpresso.cisco.com"
+                        protocol: http
+                        credentials:
+                            rest:
+                                token: <xpressoaccesstoken>
+
+        Code Example
+        ------------
+
+            >>> from pyats.topology import loader
+            >>> testbed = loader.load('testbed.yaml')
+            >>> device = testbed.devices['xpresso']
+            >>> device.connect(alias='rest', via='rest')
+        '''
+
+        if self.connected:
+            return True
+
+        if 'host' in self.connection_info:
+            host = self.connection_info['host']
+        else:
+            raise Exception("must include a host name")
+
+        if 'protocol' in self.connection_info:
+            protocol = self.connection_info['protocol']
+        else:
+            protocol = 'https'
```

Comment:
```
```suggestion
        protocol = self.connection_info.get('protocol', 'https')

```
```